src/streamlink/stream/hls.py

```diff
@@ -234,10 +234,12 @@ def reload_playlist(self):
 
         if sequences:
             self.process_sequences(playlist, sequences)
+        else:
+            self._set_playlist_reload_time(playlist, sequences)
```

Comment:
I'm not sure if this could potentially break other HLS streams. @back-to @beardypig ?

The reason I've added this here is that the default playlist reload time is 15 and if there are no segments in the first playlist "refresh" (first download), then the `_set_playlist_reload_time` method does not get called. This makes the Twitch preroll ad test wait unnecessarily long for 15 seconds.